desktop-widgets/filterwidget2.cpp

```diff
@@ -0,0 +1,118 @@
+#include "desktop-widgets/filterwidget2.h"
+#include "desktop-widgets/simplewidgets.h"
+
+#include <QDoubleSpinBox>
+
+FilterWidget2::FilterWidget2(QWidget* parent)
+: QWidget(parent)
+, ui(new Ui::FilterWidget2())
+{
+	ui->setupUi(this);
+
+	FilterData data;
+	ui->minRating->setCurrentStars(data.minRating);
+	ui->maxRating->setCurrentStars(data.maxRating);
+	ui->minVisibility->setCurrentStars(data.minVisibility);
+	ui->maxVisibility->setCurrentStars(data.maxVisibility);
+	ui->minAirTemp->setValue(data.minAirTemp);
+	ui->maxAirTemp->setValue(data.maxAirTemp);
+	ui->minWaterTemp->setValue(data.minWaterTemp);
+	ui->maxWaterTemp->setValue(data.maxWaterTemp);
+
+	// TODO: unhide this when we discover how to search for equipment.
+	ui->equipment->hide();
+	ui->labelEquipment->hide();
+	ui->invertFilter->hide();
+
+	ui->to->setDate(data.to.date());
+
+	connect(ui->maxRating, &StarWidget::valueChanged,
+		[this]{ updateFilter(); }
+	);
+
+	connect(ui->minRating, &StarWidget::valueChanged,
+		[this]{ updateFilter(); }
+	);
+
+	connect(ui->maxVisibility, &StarWidget::valueChanged,
+		[this]{ updateFilter(); }
+	);
+
+	connect(ui->minVisibility, &StarWidget::valueChanged,
+		[this]{ updateFilter();}
+	);
+
+	connect(ui->maxAirTemp, qOverload<double>(&QDoubleSpinBox::valueChanged),
```

Comment:
According to the docs, `qOverload` is >= Qt 5.7. http://doc.qt.io/qt-5/qtglobal.html#qOverload

This one might be a legitimate use of lambdas:
```[this](double){ updateFilter(); }```
with a big fat comment that Qt is trying to be as painful as possible by overloading the `valueChanged` signal with double and string parameters. [What the hell!?]

I would welcome upping the requirement to 5.7+, but this doesn't seem like a particular convincing use case. ;)